### ISSUE TITLE
Sending Token Consumes High Gas Fee

### ISSUE THREAD

**Issue by shunjizhan** [AcalaNetwork/bodhi.js#213]
when sending ACA tokens with metamask, gas fee paid is:
|                        | gasPrice (Gwei) | priority fee (Gwei) | Metamask estimation (ACA) | actual gas cost (ACA) |
|------------------------|-----------------|---------------------|---------------------------|-----------------------|
| with low priority gee  | ~200            | 1                   | 0.03 - 6.8                | 4000                  |
| with high priority gee | ~200            | ~200                | 6.7 - 6.8                 | 20                    |

need to investigate why gas cost is that high, and fix it. Ideally for the best UX, the actual gas cost should be very close the metamask estimation.

**Comment by shunjizhan** [comment]
UPDATE:
this is cause by tip too high, when inputting `maxPriorityFee = 1` in MetaMask, it actually interpret it as `1 GWEI`, which becomes `1000000000`. However, the tricky part is that since it signed the tx with `maxPriorityFee = 1000000000`, we can't divide it by 10 ^ 9 and so use `tip = 1` in extrinsic, which will get bad signature error.

**Comment by xlc** [comment]
Can you figure out how metamask come out with the numbers? Does it call some rpc calls on our endpoint? Ideally by default the priority fee should be zero.

**Comment by shunjizhan** [comment]
Currently when sending token with Metamask, the default `Max Priority Fee` will be set to the same value of `Max Fee`, this is because Metamask asks for recommended gas fees by calling 

```
https://gas-api.metaswap.codefi.network/networks/595/suggestedGasFees
```

and got nothing. So it fell back to using the default values.

And the tricky part is that, even if we can make this endpoint return what we want for `Max Priority Fee`, the result will also overwrite `Max Fee` using suggested value, instead of using our custom value from `eth_gasPrice`. 

## More References
- how Metamask fetch suggested gas prices: https://github.com/MetaMask/controllers/blob/main/src/gas/GasFeeController.ts#L384
- opened an issue on Metamask: https://github.com/MetaMask/metamask-extension/issues/13261

**Comment by xlc** [comment]
Actually the estimate fee of 6.8 ACA is fine-ish, but why the actual gas cost can be up to 5339 ACA? Max priority fee equal to Max fee is not a big issue.

**Comment by shunjizhan** [comment]
since `maxPriorityFee` is in Gwei, and 200 will become 200000000000, so our actual tip will be 200000000000 * gasLimit

**Comment by shunjizhan** [comment]
if we set priority fee to `0.00000001`, the actual cost will be very normal, like 0.03 ACA

**Comment by xlc** [comment]
Yeah I think we can somehow force metamask to use legacy format while we figuring out a better solution to this.
